pkg/discoveryservice/xdss/v2/snapshot.go

```diff
@@ -39,28 +37,28 @@ func (s Snapshot) SetResource(name string, res envoy.Resource) {
 	switch o := res.(type) {
 
 	case *envoy_api_v2.ClusterLoadAssignment:
-		s.v2.Resources[v2CacheResources(envoy_resources.Endpoint)].Items[name] = o
+		s.v2.Resources[v2CacheResources(envoy.Endpoint)].Items[name] = o
 
 	case *envoy_api_v2.Cluster:
-		s.v2.Resources[v2CacheResources(envoy_resources.Cluster)].Items[name] = o
+		s.v2.Resources[v2CacheResources(envoy.Cluster)].Items[name] = o
 
-	case *envoy_api_v2_route.Route:
-		s.v2.Resources[v2CacheResources(envoy_resources.Route)].Items[name] = o
+	case *envoy_api_v2.RouteConfiguration:
+		s.v2.Resources[v2CacheResources(envoy.Route)].Items[name] = o
```

Comment:
This was actually a bug. We never encountered it because we had never used the route object in an EnvoyConfig.